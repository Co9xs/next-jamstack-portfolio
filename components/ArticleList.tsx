import styled from 'styled-components'
import { Article } from './Article'
import { CustomLink } from './CustomLink'

export const ArticleList = ({ articles }) => {
  return (
    <ArticleListBase>
      {articles.map(article => (
          <CustomLink href={`/blog/${article.id}`} key={ article.id }>
            <ArticleListItem>
              <Article article={ article }/>
            </ArticleListItem>
          </CustomLink>
      ))}
    </ArticleListBase>
  )
}

const ArticleListBase = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
`
const ArticleListItem = styled.li`
  margin-bottom: .5rem;
`
const ArticleListLink = styled.ul`
`

