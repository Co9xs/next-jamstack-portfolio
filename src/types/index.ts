export type Category = {
  id: string,
  name: string,
  articleCount?: number
}

export type CategoryData = {
  contents: Category[],
  totalCount: number
}

export type Article = {
  id: string,
  title: string,
  body: string,
  publishedAt: string,
  category: Category | null,
  ogimage?: Image,
  tags?: Tag[],
  author?: Author
  relatedArticles: Article[]
}

export type Image = {
  url: string,
  width: number,
  height: number
}

export type Author = {
  displayName: string,
  description: string,
  image: string
}

export type Tag = {
  id: string,
  name: string
}

export type ArticleData = {
  contents: Article[],
  totalCount: number
}

export type Work = {
  title: string,
  date: string,
  url: string,
  githubUrl: string,
  languages: string[],
  description: string
}

export type Skill = {
  name: string
}

export type Layout = 'Basic' | 'SideBar'