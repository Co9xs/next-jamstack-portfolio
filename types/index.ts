export type Category = {
  name: string
}

export type Article = {
  id: string,
  title: string,
  body: string,
  description: string,
  publishedAt: string,
  category: Category | null
}