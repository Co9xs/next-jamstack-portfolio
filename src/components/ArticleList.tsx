import styled from 'styled-components'
import { ArticleCard } from '@/components'
import { Article } from '@/types'
import Link from 'next/link'

type Props = {
  articles: Article[]
}

export const ArticleList: React.VFC<Props> = ({ articles }) => {
  return (
    <ArticleListBase>
      {articles.map(article => (
        <Link href={`/blog/${article.id}`} key={ article.id }>
          <ArticleListItem key={article.id}>
            <ArticleCard article={ article }/>
          </ArticleListItem>
        </Link>
      ))}
    </ArticleListBase>
  )
}

const ArticleListBase = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
`
const ArticleListItem = styled.li`
  margin-bottom: .5rem;
`

