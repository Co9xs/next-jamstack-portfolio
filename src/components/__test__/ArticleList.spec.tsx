import { Article } from '@/types';
import React from 'react'
import renderer from 'react-test-renderer'
import { ArticleList } from '../ArticleList';

const mockArticles: Article[] = [
  {
    id: 'job-hunting-entry',
    publishedAt: '2021-04-21T08:07:07.339Z',
    title: '【テスト記事】テスト記事です',
    body: '<h2 id="h8d027c8ed3">テスト記事</h2>',
    category: {
      id: 'tech',
      name: 'Tech'
    }
  },
  {
  id: 'react-reconciliation',
  publishedAt: '2021-04-21T08:07:07.339Z',
  title: '【テスト記事2】テスト記事2です',
  body: '<h2 id="h8d027c8ed3">テスト記事2</h2>',
  category: {
    id: 'poem',
    name: 'Poem'
  }
  }
]

test('ArticleCard', () => {
  const component = renderer.create(<ArticleList articles={mockArticles}/>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot();
})