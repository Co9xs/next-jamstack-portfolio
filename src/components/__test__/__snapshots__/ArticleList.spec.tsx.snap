// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard 1`] = `
<ul
  className="sc-cTkwdZ dFPxgp"
>
  <li
    className="sc-jNMdTA cEMRPw"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <article
      className="sc-bdfBwQ cIcVTu"
    >
      <div
        className="sc-gsTCUz ywWHk"
      >
        <h3
          className="sc-dlfnbm haUCCe"
        >
          【テスト記事】テスト記事です
        </h3>
      </div>
      <div
        className="sc-hKgILt chrFSH"
      >
        <span
          className="sc-jSgupP jvTTyp"
        >
          2021/04/21
        </span>
        <span
          className="sc-eCssSg iCSNRt"
        >
          #
          Tech
        </span>
      </div>
    </article>
  </li>
  <li
    className="sc-jNMdTA cEMRPw"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <article
      className="sc-bdfBwQ cIcVTu"
    >
      <div
        className="sc-gsTCUz ywWHk"
      >
        <h3
          className="sc-dlfnbm haUCCe"
        >
          【テスト記事2】テスト記事2です
        </h3>
      </div>
      <div
        className="sc-hKgILt chrFSH"
      >
        <span
          className="sc-jSgupP jvTTyp"
        >
          2021/04/21
        </span>
        <span
          className="sc-eCssSg iCSNRt"
        >
          #
          Poem
        </span>
      </div>
    </article>
  </li>
</ul>
`;
